### ISSUE TITLE
When I run `cargo install cargo-xbuild bootimage ` in shell it happens?

### ISSUE THREAD

**Issue by ZacRenis** [phil-opp/blog_os#612]
Thanks a lot for build this OS,that let me know much about the system.
When I run `cargo install cargo-xbuild bootimage ` in shell it happens?And can you help me to fix it, or Anybody faced the same issue?  @phil-opp 

but i had install `cc`, when i `cargo new demo`,`cd demo`,`cargo run`,it works!!!!
```shell
error: failed to run custom build command for `backtrace-sys v0.1.28`

Caused by:
  process didn't exit successfully: `C:\Users\ZacRen\AppData\Local\Temp\cargo-install7Fr4BJ\release\build\backtrace-sys-33f496e80a80f631\build-script-build` (exit code: 1)
--- stdout
cargo:rustc-cfg=rbt
TARGET = Some("x86_64-pc-windows-gnu")
OPT_LEVEL = Some("3")
HOST = Some("x86_64-pc-windows-gnu")
CC_x86_64-pc-windows-gnu = None
CC_x86_64_pc_windows_gnu = None
HOST_CC = None
CC = None
CFLAGS_x86_64-pc-windows-gnu = None
CFLAGS_x86_64_pc_windows_gnu = None
HOST_CFLAGS = None
CFLAGS = None
CRATE_CC_NO_DEFAULTS = None
CARGO_CFG_TARGET_FEATURE = Some("fxsr,mmx,sse,sse2")
running: "gcc.exe" "-O3" "-ffunction-sections" "-fdata-sections" "-m64" "-I" "src/libbacktrace" "-I" "C:\\Users\\ZacRen\\AppData\\Local\\Temp\\cargo-install7Fr4BJ\\release\\build\\backtrace-sys-c1d9580aacc31dcc\\out" "-fvisibility=hidden" "-DBACKTRACE_SUPPORTED=1" "-DBACKTRACE_USES_MALLOC=1" "-DBACKTRACE_SUPPORTS_THREADS=0" "-DBACKTRACE_SUPPORTS_DATA=0" "-DHAVE_DL_ITERATE_PHDR=1" "-D_GNU_SOURCE=1" "-D_LARGE_FILES=1" "-Dbacktrace_full=__rbt_backtrace_full" "-Dbacktrace_dwarf_add=__rbt_backtrace_dwarf_add" "-Dbacktrace_initialize=__rbt_backtrace_initialize" "-Dbacktrace_pcinfo=__rbt_backtrace_pcinfo" "-Dbacktrace_syminfo=__rbt_backtrace_syminfo" "-Dbacktrace_get_view=__rbt_backtrace_get_view" "-Dbacktrace_release_view=__rbt_backtrace_release_view" "-Dbacktrace_alloc=__rbt_backtrace_alloc" 
"-Dbacktrace_free=__rbt_backtrace_free" "-Dbacktrace_vector_finish=__rbt_backtrace_vector_finish" "-Dbacktrace_vector_grow=__rbt_backtrace_vector_grow" "-Dbacktrace_vector_release=__rbt_backtrace_vector_release" "-Dbacktrace_close=__rbt_backtrace_close" "-Dbacktrace_open=__rbt_backtrace_open" "-Dbacktrace_print=__rbt_backtrace_print" "-Dbacktrace_simple=__rbt_backtrace_simple" "-Dbacktrace_qsort=__rbt_backtrace_qsort" "-Dbacktrace_create_state=__rbt_backtrace_create_state" "-Dbacktrace_uncompress_zdebug=__rbt_backtrace_uncompress_zdebug" "-o" "C:\\Users\\ZacRen\\AppData\\Local\\Temp\\cargo-install7Fr4BJ\\release\\build\\backtrace-sys-c1d9580aacc31dcc\\out\\src/libbacktrace\\alloc.o" "-c" "src/libbacktrace/alloc.c"

--- stderr


error occurred: Failed to find tool. Is `gcc.exe` installed? (see https://github.com/alexcrichton/cc-rs#compile-time-requirements for help)



warning: build failed, waiting for other jobs to finish...
error: failed to compile `cargo-xbuild v0.5.8`, intermediate artifacts can be found at `C:\Users\ZacRen\AppData\Local\Temp\cargo-install7Fr4BJ`

Caused by:
  build failed
```

**Comment by bjorn3** [comment]
```
error occurred: Failed to find tool. Is `gcc.exe` installed? (see https://github.com/alexcrichton/cc-rs#compile-time-requirements for help)
```

**Comment by AntoineSebert** [comment]
Wild guess : `cargo` can use your system's default compiler, while `cargo-xbuild` use GCC by default.

You could try to check if it is possible to pass your system's default compiler as argument to `cargo-xbuild`.

If you cannot, you can install GCC. If you already have it installed, it just means that Cargo is unable to find the compiler.

Probable installation steps:
You appear to be using a Windows operating system (`TARGET = Some("x86_64-pc-windows-gnu"`).
It is also showed that the Cargo needs GCC to build the project (`running: "gcc.exe" "-O3" "-ffunction-sections" [...]`).
Since GCC is not included in Windows operating systems by default, you have to install it manually => https://gcc.gnu.org.
But since GCC is part of the GNU collection (https://en.wikipedia.org/wiki/GNU), it does not aim to run on Microsoft products.
SO you have to provide it a runtime environment, like [MinGW](http://mingw.org/) or [Cygwin](https://www.cygwin.com/).

**Comment by phil-opp** [comment]
> Wild guess : `cargo` can use your system's default compiler, while `cargo-xbuild` use GCC by default.

That's not true. Cargo-xbuild is just a thin wrapper around cargo, it does not change any compilation options (apart from switching the sysroot).

**Comment by AntoineSebert** [comment]
But why would the triplet be `x86_64-pc-windows-gnu` in the first place ?

I think this line shows up the origin of the error:
```
error occurred: Failed to find tool. Is `gcc.exe` installed? (see https://github.com/alexcrichton/cc-rs#compile-time-requirements for help)
```
According to the description of [cc-rs](https://github.com/alexcrichton/cc-rs):
> This crate calls out to the most relevant compiler for a platform, for example using cl on MSVC.

In that case it appears to be wrong, but the `CC` flag could be used to change that.

**Comment by phil-opp** [comment]
@ZacRenis 

TLDR: I published a new version of `cargo-xbuild` that should fix this issue. Could you try to install it again?

The problem was that `cargo-xbuild` depends on [`cargo_metadata`](https://docs.rs/cargo_metadata/0.7.4/cargo_metadata/), which depends on [`error_chain`](https://docs.rs/error-chain/0.12.1/error_chain/) with the `backtrace` feature, which transitively has a dependency on [`backtrace_sys`](https://docs.rs/backtrace-sys/0.1.28/backtrace_sys/), which in turn has a build dependency on [`cc`](https://docs.rs/cc/1.0.37/cc/). The `cc` crate has a [compile-time requirements](https://github.com/alexcrichton/cc-rs#compile-time-requirements) on `gcc`.

This is unfortunate, so I published a new version of `cargo-xbuild` which doesn't use backtracing by default. Users that want the backtracking can opt-in by passing the `--features backtrace` argument.

**Comment by ZacRenis** [comment]
> @ZacRenis
> TLDR: I published a new version of cargo-xbuild that should fix this issue. Could you try to install it again?
> The problem was that cargo-xbuild depends on cargo_metadata, which depends on error_chain with the backtrace feature, which transitively has a dependency on backtrace_sys, which in turn has a build dependency on cc. The cc crate has a compile-time requirements on gcc.
> This is unfortunate, so I published a new version of cargo-xbuild (version 0.5.9) which doesn't use backtracing by default. Users that want the backtracking can opt-in by passing the --features backtrace argument.

**Comment by ZacRenis** [comment]
Yes, it works,Thanks.
